packages/server/api/controllers/reviews.controller.js

```diff
@@ -0,0 +1,38 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+
+const getAllReviews = async () => {
+  try {
+    const allReviews = await knex.select().table('reviews');
+
+    if (allReviews.length === 0) {
+      throw new HttpError(`No reviews`, 404);
+    }
+    return allReviews;
+  } catch (error) {
+    return error.message;
+  }
+};
+
+const getReviewsByProductId = async (productId) => {
+  if (!productId) {
```

Comment:
This won't really work, what if I pass in 0. 0 is a number but it's falsy so this check will throw. 
Also note that in HTTP all parameters are strings, so you need to convert it to a number first. 

Roughly, you should combine a "not empty" check with a `parseInt` check.